[pickers] Update `closeOnSelect` and `actionBar.actions` default values (#15944)
mui/mui-x: docs/data/date-pickers/lifecycle/lifecycle.md

#### When the last view is completed

When a selection in the last view is made, `onClose` will be called only if the `closeOnSelect` prop is equal to `true`.
The default value of `closeOnSelect` depends on the component:

- Date Picker and Date Range Picker: `true` on desktop and `false` on mobile variants;

> - Time Picker, Date Time Picker, and Date Time Range Picker: `false` on desktop and mobile variants.

Reviewer: ```suggestion
- Time Picker, Date Time Picker and Date Time Range Picker: `false` on desktop and mobile variants.
```

Author: These were incorrect suggestions.
Time Picker, Date Time Picker and Date Time Range Picker: `false` on desktop and mobile variants.
![Screenshot 2024-12-30 at 16 55 04](https://github.com/user-attachments/assets/58960e1c-4679-4408-8012-6e333ab749c5)

Reviewer 2: https://www.grammarly.com/blog/punctuation-capitalization/comma-before-and/

This seems to be a complex topic :laughing:

Author: Ok... I didn't read that blog post before...
It seems that both variants are correct, but I think that we mostly use `,` before the "3rd" `and`.

> - Time Picker, Date Time Picker and Date Time Range Picker: `false` on both desktop and mobile variants.

Author: ```suggestion
- Time Picker, Date Time Picker, and Date Time Range Picker: `false` on desktop and mobile variants.
```

  ```

  - Default `views` prop: `['hours', 'minutes']` (plus a `meridiem` view if the locale is in 12-hours format)
  - Default `closeOnSelect` prop: `false`

>   **Behavior:** The picker won't close when selecting the minutes or meridiem (if a 12-hour clock is used).

Reviewer 2: Nitpick

```suggestion
  **Behavior:** The picker will not close when selecting the minutes or meridiem (if a 12-hour clock is used).
```

It seems that we are using both but "will not" a lot more than "won't"

Author: Replaced all cases of `won't` in this file.

#### When the last view is completed

When a selection in the last view is made, `onAccept` will be called only if the `closeOnSelect` prop is equal to `true` and the value has been modified since the last time `onAccept` was called.
The default value of `closeOnSelect` depends on the component used:

- Date Picker and Date Range Picker: `true` on desktop and `false` on mobile variants;

> - Time Picker, Date Time Picker, and Date Time Range Picker: `false` on desktop and mobile variants.

Reviewer: ```suggestion
- Time Picker, Date Time Picker and Date Time Range Picker: `false` on desktop and mobile variants.
```
